tests/foreman/ui/test_repository.py

```diff
@@ -339,6 +338,24 @@ def test_positive_discover_module_stream_repo_via_existing_product(session, modu
 
     :BZ: 1676642
     """
+    repo_name = gen_string('alpha')
+    repo_label = gen_string('alpha')
+    product = entities.Product(organization=module_org).create()
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.product.discover_repo(
+            {
+                'repo_type': 'Yum Repositories',
+                'url': CUSTOM_MODULE_STREAM_REPO_2,
+                'discovered_repos.repos': "/",
+                'create_repo.product_type': 'Existing Product',
+                'create_repo.product_content.product_name': product.name,
+                'create_repo.create_repos_table': [
+                    {"Repository Name": repo_name, "Repository Label": repo_label}
+                ],
+            }
+        )
+        assert repo_name in session.repository.search(product.name, repo_name)[0]['Name']
```

Comment:
Wouldn't it make sense to assert both name and label? Since you create both

Reply:
@mirekdlugosz Thanks for the suggestion, here to validate this test asserting repo_name would be enough(referred some tests above), as a label can be generated with the repo_name field too, and here I used the label just to make a difference between names and labels.